src/reducer.js

```diff
@@ -89,27 +118,50 @@ export const actionReducers = {
         const parentObject = _.get(newValue, parentPath)
 
         if (parentObject && _.includes([null, ''], value)) {
+          valueChangeSet.set(bunsenId, {
+            value,
+            type: 'unset'
+          })
           newValue = unset(newValue, bunsenId)
         } else {
+          valueChangeSet.set(bunsenId, {
+            value,
+            type: 'set'
+          })
           newValue = set(newValue, bunsenId, value)
         }
       } else if (_.includes([null, ''], value) || (Array.isArray(value) && value.length === 0)) {
+        valueChangeSet.set(bunsenId, {
+          value,
+          type: 'unset'
+        })
         newValue = unset(newValue, bunsenId)
       } else {
-        newValue = set(newValue, bunsenId, value)
+        if (!_.isEqual(value, _.get(newValue, bunsenId))) {
+          valueChangeSet.set(bunsenId, {
+            value,
+            type: 'set'
+          })
+          newValue = set(newValue, bunsenId, value)
+        }
       }
     }
-    const newModel = evaluateConditions(state.baseModel, newValue)
-    let model
-    if (!_.isEqual(state.model, newModel)) {
-      model = newModel
-    } else {
-      model = state.model
+
+    let model = state.model || state.baseModel
+    if (valueChangeSet.size > 0) {
+      const newModel = evaluateConditions(_.cloneDeep(state.baseModel), newValue)
+      if (!_.isEqual(state.model, newModel)) {
+        model = newModel
+      } else {
+        model = state.model
+      }
```

Comment:
I'd be tempted to do `model = _.isEqual(state.model, newModel) ? state.model : newModel` here.